nav2_costmap_2d/src/footprint_collision_checker.cpp

```diff
@@ -127,13 +127,14 @@ void FootprintCollisionChecker<CostmapT>::setCostmap(CostmapT costmap)
 
 template<typename CostmapT>
 double FootprintCollisionChecker<CostmapT>::footprintCostAtPose(
-  double x, double y, double theta, const Footprint footprint)
+  double x, double y, double theta, const Footprint & footprint)
 {
   double cos_th = cos(theta);
   double sin_th = sin(theta);
   Footprint oriented_footprint;
+  oriented_footprint.reserve(footprint.size());
+  geometry_msgs::msg::Point new_pt;
   for (unsigned int i = 0; i < footprint.size(); ++i) {
-    geometry_msgs::msg::Point new_pt;
```

Comment:
The `.reserve` makes sense. But I was under the impression moving `new_pt` out of the loop only introduced potential scoping issues. The compiler would optimize this away.

Happy to learn though.

Reply:
That change did not have measurable changes, I previously had everything inlined into the `oriented_footprint` with `emplace_back` looking to avoid a copy. That didn't to anything so I aired on the side of reverting changes if they didn't help. This could be reverted back as well

Reply:
Honestly, none of the changes here really made much of a difference. The collision checker is still a nearly totally unexplored area of improvements. The cost critic still takes between 0.4-0.7 ms (which is 10-100x more than most others) of which that time is primarily spent in the collision checking process. 

I intended to spend more time to really dig into that, but I found other changes dropped the ~2ms from before down to the 0.4-0.7 range which I felt was "good enough". I think there's more room to squeeze out performance with a specialized MPPI collision checker object with precomputation / caching / operating on `float`s.